test/api/v3/integration/groups/POST-groups_groupId_join.js

```diff
@@ -0,0 +1,51 @@
+import {
+  generateUser,
+} from '../../../../helpers/api-integration.helper';
+
+describe('POST /group/:groupId/join', () => {
+  let user;
+
+  beforeEach(() => {
+    return generateUser().then((generatedUser) => {
+      user = generatedUser;
+    });
+  });
+
+  context('Guilds', () => {
+    it('inviting to guild doesn\'t give basilist to inviter', async () => {
+      let groupName = 'Test Guild';
+      let groupType = 'guild';
+
+      user = await generateUser({balance: 1, 'items.quests.basilist': 0 })
+      let guild = await user.post('/groups', {
+        name: groupName,
+        type: groupType
+      });
+
+      let invitedUser = await generateUser({ 'invitations.guilds': [{ id: guild._id}]});
+      await invitedUser.post('/groups/' + guild._id + '/join');
+      
+      let changedUser = await user.get('/user')
+      expect(changedUser.items.quests.basilist).to.equal(0);
+    });
+  });
+
+  context('Parties', () => {
+    it('gives basilist quest item to the inviter', async () => {
+      let groupName = "Test Party";
+      let groupType = "party";
+
+      user = await generateUser({'items.quests.basilist': 0 });
+      let party = await user.post('/groups', {
+        name: groupName,
+        type: groupType
+      })
+      
+      let invitedUser = await generateUser({ 'invitations.party': { id: party._id, inviter: user._id }} )
+      await invitedUser.post('/groups/' + party._id + '/join');
+      
+      let changedUser = await user.get('/user');
+      expect(changedUser.items.quests.basilist).to.equal(1);

```

Comment:
For things like this, I'm unsure of which syntax we want to use. We could do it like you have it, or we could do it like this:

```js
await expect(user.get('/user')).to.eventually.have.deep.property('items.quest.basilist', 1);
```

Does anyone have thoughts?